Cannot build xchat locally:

```
gcc -DHAVE_CONFIG_H -I. -I../.. -I/nix/store/6qbv9vdygxwzwhcbw1kimd76vhl6hiv6-glib-2.46.2/include/glib-2.0 -I/nix/store/6qbv9vdygxwzwhcbw1kimd76vhl6hi
v6-glib-2.46.2/lib/glib-2.0/include    -g -O2 -Wall -pipe -Wno-pointer-sign -funsigned-char  -c -o cfgfiles.o cfgfiles.c
In file included from xchat.h:3:0,
                 from cfgfiles.c:27:
/nix/store/6qbv9vdygxwzwhcbw1kimd76vhl6hiv6-glib-2.46.2/include/glib-2.0/glib/gslist.h:29:2: error: #error "Only <glib.h> can be included directly."
 #error "Only <glib.h> can be included directly."
  ^
```

While hexchat builds fine.

Also please for the commit message use something like this according to our policy: `hexchat: fix enchant depenency usage at runtime`

--- a/pkgs/applications/networking/irc/hexchat/default.nix
+++ b/pkgs/applications/networking/irc/hexchat/default.nix
@@ -20,6 +20,11 @@ stdenv.mkDerivation rec {
     desktop_file_utils hicolor_icon_theme
   ];
 
+ #hexchat and heachat-text loads enchant spell checking library at run time and so it needs to have route to the path
+  patchPhase = ''
+    sed -i "s,libenchant.so.1,${enchant}/lib/libenchant.so.1,g" src/fe-gtk/sexy-spell-entry.c
+  '';
+
   configureFlags = [ "--enable-shm" "--enable-textfe" ];
 
   meta = with stdenv.lib; {

--- a/pkgs/applications/networking/irc/xchat/default.nix
+++ b/pkgs/applications/networking/irc/xchat/default.nix
@@ -1,4 +1,4 @@
-{stdenv, fetchurl, pkgconfig, tcl, gtk}:
+{stdenv, fetchurl, pkgconfig, tcl, gtk, gtkspell }:
 
 stdenv.mkDerivation {
   name = "xchat-2.8.8";
@@ -6,8 +6,8 @@ stdenv.mkDerivation {
     url = http://www.xchat.org/files/source/2.8/xchat-2.8.8.tar.bz2;
     sha256 = "0d6d69437b5e1e45f3e66270fe369344943de8a1190e498fafa5296315a27db0";
   };
-  buildInputs = [pkgconfig tcl gtk];
-  configureFlags = "--disable-nls";
+  buildInputs = [pkgconfig tcl gtk gtkspell];
+  configureFlags = "--disable-nls --enable-spell=gtkspell";
 
   patches = [ ./glib-top-level-header.patch ];
 

